gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java

```diff
@@ -70,7 +70,7 @@ public Value doConvert(String value, AttributeKind attributeKind, Value.Builder
 
       case TYPE_TIMESTAMP:
         valueBuilder.setValueType(ValueType.TIMESTAMP);
-        valueBuilder.setTimestamp(Long.parseLong(value));
+        valueBuilder.setTimestamp(Double.valueOf(value).longValue());
```

Comment:
what is the reason for changing this?

Reply:
What's being stored in scientific notation? That sounds like a bug somewhere - especially because this hasn't historically been an issue. If we expect to represent it as a long, it should be stored such, since we're losing precision here.